src/tcache.c

```diff
@@ -800,7 +809,9 @@ tcache_bin_info_compute(cache_bin_info_t tcache_bin_info[TCACHE_NBINS_MAX]) {
 	 * than tcache_nbins, no items will be cached.
 	 */
 	for (szind_t i = 0; i < TCACHE_NBINS_MAX; i++) {
-		unsigned ncached_max = tcache_ncached_max_compute(i);
+		unsigned ncached_max = opt_bin_info_is_set[i] ?
+		    opt_tcache_bin_info[i].ncached_max:
+		    tcache_ncached_max_compute(i);
```

Comment:
Let's avoid the redundant computation for all threads. This part can be moved to tcache_boot with the results committing into `opt_tcache_bin_info`. Then here we can simply use the results. Worth converting this into asserts though (i.e. equals to _max_comput() if not set).

Reply:
Refactored the init so that this function will only be called in `tcache_boot` and now all auto tcache inits are done through `tsd_tcache_data_init_with_bin_settings`